src/main/java/picard/nio/PicardHtsPath.java

```diff
@@ -117,4 +119,31 @@ public static List<Path> toPaths(final Collection<PicardHtsPath> picardHtsPaths)
         Objects.requireNonNull(picardHtsPaths);
         return picardHtsPaths.stream().map(PicardHtsPath::toPath).collect(Collectors.toList());
     }
+
+    /**
+     * Instead of a static method in PicardHtsPath, it could very well be a static method in PicardIOUtils.
+     *
+     * Examples:
+     *     - test_na12878.bam -> test_na12878.bai (append = false)
+     *     - test_na12878.bam -> test_na12878.bam.md5 (append = true)
+     *
+     * @param path the original path
+     * @param append whether to append (true) or replace (false) the new extension
+     * @param newExtension the extension including the dot e.g. ".txt"
+     * @return a new PicardHtsPath object pointed to a file with
+     */
+    public static PicardHtsPath replaceExtension(final PicardHtsPath path, final String newExtension, final boolean append){
+        if (append){
+            return new PicardHtsPath(path.getURIString() + newExtension);
+        } else {
+            return new PicardHtsPath(FilenameUtils.removeExtension(path.getURI().toString() + newExtension));
```

Comment:
Thanks for catching this. Now using resolveSibling. Will add some tests and see if FilnameUtils can be replaced by something in house. Let's discuss whether we should move this to htsjdk during next week's meeting.

Reply:
Replaced FileNameUtils::removeExtension with a combination of IOPath::getExtension() and String::replaceAll(). Also added tests.

Reply:
As for moving this method to resolve, I think replaceExtension() belongs in PicardHtsPath because it returns a new PicardHtsPath object (much like PicardHtsPath::fromPath()). If it's in HtsPath we would have to override it, right?